File: Oving 1/oppg4.py
# Python 3.3.3 and 2.7.6
# python helloworld_python.py

from threading import Thread

i = 0

def ThreadFunction_1():
    for n in range (0,1000000):
        global i 
        i -= 1


def ThreadFunction_2():
    for n in range (0,1000000):
        global i 
        i += 1

# Potentially useful thing:
#   In Python you "import" a global variable, instead of "export"ing it when you declare it
#   (This is probably an effort to make you feel bad about typing the word "global")

def main():
    Thread_1 = Thread(target = ThreadFunction_1, args = (),)
    Thread_2 = Thread(target = ThreadFunction_2, args = (),)
    Thread_1.start()
    Thread_2.start()
    
    Thread_1.join()
    Thread_2.join()
    print(i)

main()

File: Oving 1/oppg4.cpp
//Compiler:
//g++ -std=c++11 -pthread file_name.cpp
	
#include<iostream>
#include <thread>


using namespace std;

static int i = 0;

void threadfunction_1() {
	for (int n = 0; n < 1000000; n++) {
		i++;
	}
}

void threadfunction_2() {
	for (int n = 0; n < 1000000; n++){
		i--;
	}
}

int main() {
	thread thread_1(threadfunction_1);
	thread thread_2(threadfunction_2);
	thread_1.join();
	thread_2.join();
	cout << i << endl;
	return 0;
}
